backend/analysis-engine/src/main/java/org/sonarsource/sonarlint/core/analysis/AnalysisQueue.java

```diff
@@ -85,12 +85,16 @@ private Optional<QueuedCommand> pollNextReadyCommand() {
     // cannot use iterator as priority order is not guaranteed
     while (!queue.isEmpty()) {
       var candidateCommand = queue.poll();
-      if (candidateCommand.command.isReady()) {
-        queue.addAll(commandsToKeep);
-        return Optional.of(candidateCommand);
+      if (candidateCommand.command.isCanceled()) {
+        LOG.debug("Not picking next command {}, is canceled", candidateCommand.command);
```

Comment:
I think you have to call `Command.cancel` at some point, for the analysis future to be canceled. It shows that the design is not great, because if the command is canceled, we should call cancel.... At some point we should maybe have a whiteboard session. Another option is to rely on `SonarLintCancelMonitor.onCancel`.